src/api/java/com/minecolonies/api/tileentities/TileEntityStash.java

```diff
@@ -0,0 +1,74 @@
+package com.minecolonies.api.tileentities;
+
+import com.minecolonies.api.colony.buildings.IBuilding;
+import com.minecolonies.api.colony.buildings.IBuildingContainer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntityType;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nonnull;
+
+import static com.minecolonies.api.util.constant.BuildingConstants.MAX_PRIO;
+import static com.minecolonies.api.util.constant.Constants.DEFAULT_SIZE;
+
+/**
+ * Class which handles the tileEntity for the Stash block.
+ */
+
+public class TileEntityStash extends TileEntityColonyBuilding
+{
+
+    public TileEntityStash(final TileEntityType type)
+    {
+        super(type);
+        inventory = new NotifyingRackInventory(DEFAULT_SIZE);
+    }
+
+    public TileEntityStash()
+    {
+        super(MinecoloniesTileEntities.STASH);
+        inventory = new NotifyingRackInventory(DEFAULT_SIZE);
+    }
+
+    /**
+     * Called when the inventory of the tileEntity it holds it's changed
+     *
+     * @param isEmpty
+     */
+    private void buildingInventoryChanged(boolean isEmpty)
+    {
+        IBuildingContainer building = getBuilding();
+        if (!isEmpty && building != null && !building.isPriorityStatic())
+        {
+            if (building instanceof IBuilding)
+            {
+                IBuilding iBuilding = (IBuilding) building;
+                if (!iBuilding.isBeingGathered())
+                {
+                    iBuilding.alterPickUpPriority(MAX_PRIO);
+                }
+            } else
```

Comment:
else on other line

Reply:
Sorry, I thought I had the auto-formatter fixed this. I'm fixing this manually.